Source/WebCore/editing/Editor.cpp

```diff
@@ -1159,6 +1159,8 @@ void Editor::appliedEditing(CompositeEditCommand& command)
         OptionSet<FrameSelection::SetSelectionOption> options;
         if (command.isDictationCommand())
             options.add(FrameSelection::DictationTriggered);
+        if (isProcessingKeyCommand())
```

Comment:
Could this method be defined on `CompositeEditCommand` rather than on `Editor`? Similar to `isDictationCommand()` above, I wonder if this check could be `command.isKeyCommand()`.

cc @whsieh

Reply:
I think having the state on `CompositeEditCommand` would be nice, but it may be tricky, since what we really care about is "was the `Editor::Command` created with `CommandFromMenuOrKeyBinding`".

Reply:
That aside, I think there's (broadly speaking) two viable strategies we can take, to fix the bug at hand:
• Make selection changes during user-triggered edit commands actually use the `UserTriggered` flag when setting the selection.
• ...or, make it so that scheduling an async selection change never results in us painting a stale selection caret.

It seems like it might be worth investigating the second bullet a bit, to figure out why the async selection change leaves behind this painting artifact. I suspect this investigation would also give us some insight as to how we can devise a test to exercise this bug.

(It might have something to do with installing zero-delay timers that race against the next rendering update, after performing an "InsertParagraph" command)